ship.js

```diff
@@ -523,17 +522,15 @@ class Plane extends Entity {
         // handle powerUp grab by player
         if (entity.payload.powerUp && !entity.isPlayer) {
           // TODO: store powerUps for user activation and update the HUD inventory
-          console.log('Calling the power up');
-          // console.log(`Lives is ${this.pa}`)
-          entity.payload.powerUp(this); // for now just run the enclosed powerUp
 
-          console.log('Done callign the power up;');
-          entity.removeFromWorld = true;
-        } else {
-          // hit by enemy bullet
+          entity.payload.powerUp(this); // for now just run the enclosed powerUp
+          // entity.removeFromWorld = true;
+        } else if (this.shield.hasShield) {
```

Comment:
This is the update to the player's hit detection that checks to see if a shield is present before removing a life